homeassistant/components/geo_location/__init__.py

```diff
@@ -64,32 +71,38 @@ def state(self):
     @property
     def source(self) -> str:
         """Return source value of this external event."""
+        if hasattr(self, "_attr_source"):
+            return self._attr_source
         raise NotImplementedError
 
     @property
     def distance(self) -> float | None:
         """Return distance value of this external event."""
+        if hasattr(self, "_attr_distance"):
+            return self._attr_distance
         return None
```

Comment:
Moved the default to the class attribute.

```suggestion
        return self._attr_distance
```